### ISSUE TITLE
RibbonContextualTabGroup.Visiblity binding not work.

### ISSUE THREAD

**Issue by GuangZhouDekang** [fluentribbon/Fluent.Ribbon#905]
### Environment

- Fluent.Ribbon  8.0.3
- Windows  10
- .NET Framework  4.5.2

I set the contextualgroups into binding an bool value. but it not working.

`<fluent:Ribbon.ContextualGroups>
                <fluent:RibbonContextualTabGroup x:Name="tooldatabaseGroup"
                                                 Header="tooldatabase"
                                                 Visibility="{Binding IsTabToolDatabaseSelected,Converter={StaticResource BooleanToVisibilityConverter}}"
                                                 Background="Green"
                                                 BorderBrush="Green" />
                 
            </fluent:Ribbon.ContextualGroups>`

 the Convert method is not called. I checked the converter, it work right and can call in ther other xaml.

**Comment by GuangZhouDekang** [comment]
If I use the lower version Fluent.Ribbon 8.0.0
It work right.
Is the new version not Bindable?

**Comment by batzen** [comment]
Will have a look if i broke that.

**Comment by batzen** [comment]
I can't reproduce your issue with the showcase application.
Could you add a repro for your issue?

**Comment by pisc-software** [comment]
I am having the same issue here, using MahApps with Microsoft Community Toolkit MVVM. The binding is not being initialized (called when the view is generated) and also not updated when triggering PropertyChanged. Context Tabs are showing when setting Visbility without binding.

`            
            <fluent:Ribbon.ContextualGroups>
                <fluent:RibbonContextualTabGroup Header="This tab shows" x:Name="contextGroupTabShows"
                                                 Visibility="Visible"/>

                <fluent:RibbonContextualTabGroup Header="This tab does not show" x:Name="contextGroupTabNotShowing"
                                                 Visibility="{Binding TabSelectedItem.IsShowContext, Mode=OneWay, Converter={StaticResource booleanToVisibilityConverter}}" />
            </fluent:Ribbon.ContextualGroups>
`

**Comment by batzen** [comment]
@pisc-software Your Window should implement `IRibbonWindow` as shown in https://github.com/fluentribbon/Fluent.Ribbon/blob/develop/Fluent.Ribbon.Showcase/MahMetroWindow.xaml.cs
I just improved that showcase with d76607e242443a8e6a7117befe2481fe04ecef07 as i noticed that bindings in the title bar didn't work as intended.

**Comment by pisc-software** [comment]
Thanks for the reply.

For the binding issue that is what I did debug so far:

It seems in my implementation that the DataContext does not propagate to "Ribbon.ContextualGroups". Setting the DataContext for the UserControl (in my case) does work for the UserControl, but it does not propagate to the Contextual Group.
When extending a control class from the "RibbonContextualTabGroup" class and inspecting the DataContext in the constructor the DataContext set for the UserControl (or the Ribbon, or even the RibbonContextualTabGroup in XAML) is not there.

```
    <UserControl.DataContext>
        <vm:MainViewModel />
    </UserControl.DataContext>

```
```
            <control:FluentRibbonContextualTabGroup Header="Some Header" x:Name="contextGroup"
              Visibility="{Binding TabSelectedItem.IsContext, Converter={StaticResource booleanToVisibilityConverter}}"
            >
                <control:FluentRibbonContextualTabGroup.DataContext>
                    <vm:MainViewModel />
                </control:FluentRibbonContextualTabGroup.DataContext>
            </control:FluentRibbonContextualTabGroup>

```

So as workaround the DataContext needs to be set in the child control class:

```
    public class FluentRibbonContextualTabGroup : Fluent.RibbonContextualTabGroup
    {
        public FluentRibbonContextualTabGroup() : base()
        {
            this.DataContext = App.Current.Services.GetRequiredService<MainViewModel>();
        }
```
Doing that makes the bindings work. In my case that workaround is so far workable, as also additional attached properties are used for the ContextualTab for visual control.

My use case uses IoC from the Community Toolkit MVVM.
Any thoughts on this?

**Comment by batzen** [comment]
The `DataContext` is not yet set when the object is constructed in WPF, so it's normal that when you debug it's null at that point in time.

As i said in my previous response you could set the datacontext for your `RibbonTitleBar` as i did in the showcase application.
Like this:
```
    <Metro:MetroWindow.TitleTemplate>
        <DataTemplate>
            <Fluent:RibbonTitleBar x:Name="RibbonTitleBar"                                   
                                   Header="{Binding Path=Title, RelativeSource={RelativeSource AncestorType=Window}}"
                                   DataContext="{Binding ElementName=YOUR_ELEMENT_NAME, Path=DataContext}" />
        </DataTemplate>
    </Metro:MetroWindow.TitleTemplate>
```

**Comment by pisc-software** [comment]
Thank you. That actually was the missing part. Binding now works also in my case here.

I was not aware that it is needed to set the DataContext for the Ribbon through the RibbonTitleBar, and I did not fully see the relevance in your code snippet right away.

I did also use the XAML debugger ([https://marketplace.visualstudio.com/items?itemName=PeterSpa.XamlBinding]) also for tracking the binding, and there also the binding in question was shown as unresolvable with an unset DataContext. Also when using the live visual tree the DataContext of the ContextTabGroup was null. That did lead me to extending the class to see what happens and what a possible approach could be.

Thanks for the response and all the beautiful work on this. I hope that anyone stumbling across this will find this thread also helpful.
